packages/vscode-dependencies-validation/src/autofix/depsIssues.ts

```diff
@@ -0,0 +1,59 @@
+import type { Uri } from "vscode";
+import {
+  VscodeFileEventConfig,
+  VscodeUriFile,
+  VscodeWorkspace,
+} from "../vscodeTypes";
+import {
+  ENABLE_AUTOFIX,
+  getAutoFixDelay,
+  isAutoFixEnabled,
+} from "./configuration";
+import { addPackageJsonFileWatcher as addPackageJsonFileWatcher } from "./packageJsonFileWatcher";
+import { addUnsupportedFilesWatcher } from "./unsupportedFilesWatcher";
+import { findAndFixDepsIssues } from "../util";
+import { clearDiagnostics } from "../util";
+
+export function activateDepsIssuesAutoFix(
+  vscodeConfig: VscodeFileEventConfig & VscodeUriFile
+): void {
+  fixWorkspaceDepsIssues(vscodeConfig);
+  addPackageJsonFileWatcher(vscodeConfig);
+  addUnsupportedFilesWatcher(vscodeConfig);
+
+  onAutoFixChange(vscodeConfig);
+}
+
+function fixWorkspaceDepsIssues(vscodeConfig: VscodeFileEventConfig): void {
+  setTimeout(() => {
+    void findDepsIssuesFixThemAndCleanProblemsView(vscodeConfig);
+  }, getAutoFixDelay(vscodeConfig.workspace));
+}
+
+function getPackageJsonUris(workspace: VscodeWorkspace): Thenable<Uri[]> {
+  return workspace.findFiles("package.json", "**​/node_modules/**");
```

Comment:
random thought:
is there a value in filtering?
- edge case: a package.json file which does not meet the schema.
- edge case: a package.json which has no deps and dev-deps --> can be ignored.
- optimization: knowing for each package.json if it has deps/devDeps/both for our `npm ls` flows.

Reply:
I did not understand what you tried to improve here.
Let's talk about it.